src/ImageSharp/Configuration.cs

```diff
@@ -81,19 +115,53 @@ public Configuration(params IImageFormat[] providers)
 #endif
 
         /// <summary>
-        /// Adds a new <see cref="IImageFormat"/> to the collection of supported image formats.
+        /// Registers a new format provider.
+        /// </summary>
+        /// <param name="formatProvider">The format providers to call configure on.</param>
+        public void AddImageFormat(IImageFormatProvider formatProvider)
+        {
+            Guard.NotNull(formatProvider, nameof(formatProvider));
+            formatProvider.Configure(this);
+        }
+
+        /// <inheritdoc />
+        public void SetMimeTypeEncoder(string mimeType, IImageEncoder encoder)
```

Comment:
I disagree it should be fully public and discoverable otherwise you can't do :
```c#
Configuration.Default.SetMimeTypeEncoder("image/jpeg", new JpegEncoder()
{
    Quality = 90,
    IgnoreMetadata = true
});
```